packer/registry/image/image.go

```diff
@@ -50,10 +50,6 @@ func (i *Image) Validate() error {
 		return errors.New("error registry image does not contain a valid ProviderName")
 	}
 
-	if i.SourceImageID == "" {
-		return errors.New("error registry image does not contain a valid SourceImageID")
-	}
-
```

Comment:
Since this is optional, I think this should not be an error when this is not passed. Also, after taking a look around plugins, sometimes this was wrongly used, as in for the amazon plugin, the ami was used 🤔.

Reply:
I think it would be a good idea to have the image.`cloud_source_image_id` AND the build.`hcp_source_image_id` though.

Reply:
The API is supposed to get this AMI id and find the source image that has it as ID. So in HCP, this will be transformed into HCP-related data.

Reply:
But wait, in the amazon plugin, which AMI was used?

Reply:
Here: https://github.com/hashicorp/packer-plugin-amazon/blob/998d3209616d413eff1847c1bd7e5909b7b3c96c/builder/common/artifact.go#L167-L169

Reply:
Ahh ok, so that's right! The source AMI is set with the one queried from hcp, if using the data source. So the API will know who's the parent from it. 🤔

Reply:
Gotcha !